hydra-node/src/Hydra/Options.hs

```diff
@@ -305,22 +305,33 @@ chainConfigParser =
 
 networkIdParser :: Parser NetworkId
 networkIdParser =
-  testnetParser
+  option
+    (parseTestnet <|> parseMainnet)
+    ( long "network-id"
+        <> showDefault
+        <> value (Testnet (NetworkMagic 42))
+        <> completer (listCompleter ["1", "2", "42", "mainnet", "Mainnet", "m"])
+        <> help
+          "Either Mainnet network or a magic number identifying the testnet to connect to. We only need to \
+          \provide the magic number here. For example: '2' is the 'preview' \
+          \network. See https://book.world.dev.cardano.org/environments.html for available networks."
```

Comment:
Not sure if it was implemented that way, but we could align this command line option with the one from `cardano-cli` https://github.com/input-output-hk/cardano-node/issues/4418

Could (definitely not required, happy to discuss):
* drop `--network-id` (which was only about the magic)
* add `[--mainnet | --testnet-magic]`

Reply:
I tried to be _backwards compatible_ and not break things but I can definitely use parser for `cardano-cli` for this too.